ldpd/lde.c

```diff
@@ -1202,6 +1216,89 @@ lde_send_labelrelease(struct lde_nbr *ln, struct fec_node *fn,
 	lde_imsg_compose_ldpe(IMSG_RELEASE_ADD_END, ln->peerid, 0, NULL, 0);
 }
 
+void
+lde_send_labelrequest(struct lde_nbr *ln, struct fec_node *fn,
+		      struct map *wcard, int single)
+{
+	struct map		 map;
+	struct fec		*f;
+	struct lde_req		*lre;
+
+	if (fn) {
+		lde_fec2map(&fn->fec, &map);
+		switch (fn->fec.type) {
+		case FEC_TYPE_IPV4:
+			if (!ln->v4_enabled)
+				return;
+			if (lde_acl_check(ldeconf->ipv4.acl_label_advertise_to,
+				AF_INET, (union ldpd_addr *)&ln->id, 32)
+				!= FILTER_PERMIT)
+				return;
+			if (lde_acl_check(ldeconf->ipv4.acl_label_advertise_for,
+				AF_INET,
+				(union ldpd_addr *)&fn->fec.u.ipv4.prefix,
+				fn->fec.u.ipv4.prefixlen) != FILTER_PERMIT)
+				return;
+			break;
+		case FEC_TYPE_IPV6:
+			if (!ln->v6_enabled)
+				return;
+			if (lde_acl_check(ldeconf->ipv6.acl_label_advertise_to,
+				AF_INET, (union ldpd_addr *)&ln->id, 32)
+				!= FILTER_PERMIT)
+				return;
+			if (lde_acl_check(ldeconf->ipv6.acl_label_advertise_for,
+				AF_INET6,
+				(union ldpd_addr *)&fn->fec.u.ipv6.prefix,
+				fn->fec.u.ipv6.prefixlen) != FILTER_PERMIT)
+				return;
+			break;
+		default:
+			fatalx("lde_send_labelrequest: unknown af");
```

Comment:
This looks like it just crashes out -- should we crash here, or just print an error/log and ignore this input? Just wondering about this being the right result here.

Reply:
The way this function is coded matches the rest of the LDP code.   I will leave it up to Renato if wants me to change this code.

Reply:
I think this is fine. This kind of error would correspond to an internal inconsistency that isn't worth trying to recover from.